eva/experimental/ray/executor/ray_stage.py

```diff
@@ -15,28 +15,32 @@
 from typing import Callable, List
 
 import ray
+from ray.exceptions import RayError
 from ray.util.queue import Queue
 
+from eva.executor.executor_utils import ExecutorError
+
 
 class StageCompleteSignal:
     pass
 
 
 @ray.remote(num_cpus=0)
 def ray_stage_wait_and_alert(tasks: ray.ObjectRef, output_queue: Queue):
-    ray.get(tasks)
-    for q in output_queue:
-        q.put(StageCompleteSignal)
+    try:
+        ray.get(tasks)
+        for q in output_queue:
+            q.put(StageCompleteSignal)
+    except RayError as e:
+        for q in output_queue:
+            q.put(ExecutorError(e.cause))
```

Comment:
Do need to handle `ExecutorError` in  `QueueReaderExecutor`? For example, for `StageCompleteSignal`, it needs to be put back into the input queue, so that other readers can also be notified.

Reply:
You are right. It needs to be populated to other queues as well.